gadget/layout.go

```diff
@@ -0,0 +1,214 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package gadget
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"sort"
+)
+
+type LayoutConstraints struct {
```

Comment:
Naming nitpick: to avoid overloading the term "layout" could we instead use the word "positioning" here? PositioningConstraints, PositionVolume, etc.

Reply:
Oh yes, that's a good point, +1 for a different name

Reply:
+1 for that, it's always good to avoid ambiguities in terminology.

Reply:
Updated type/func names and some error messages.

Reply:
There is still a lot of "layout" wording (layout.go, layoutTestSuite, tests).

Reply:
Heh, I can do that too. Will break a bit my work-in-progress branch, but I can work around that probably.